mlrun/serving/routers.py

```diff
@@ -157,3 +163,256 @@ def _handle_event(self, event):
+        return model, self.routes[model], subpath
+
+    def _max_vote(self, all_predictions):
+        """Returns an argmax"""
+        return [
+            max(predictions, key=predictions.count) for predictions in all_predictions
+        ]
+
+    def _mean_vote(self, all_predictions):
+        return [mean(predictions) for predictions in all_predictions]
+
+    def _is_int(self, n):
+        return float(n).is_integer()
+
+    def _vote(self, events):
+        if "outputs" in events.body:
+            # Dealing with a specific model prediction
+            return events
+        predictions = [
+            model.body["outputs"]["prediction"] for _, model in events.body.items()
+        ]
+
+        flattened_predictions = [
+            [predictions[j][i] for j in range(len(predictions))]
+            for i in range(len(predictions[0]))
+        ]
+        if self.vote_flag is None:
+            if all(
+                [
+                    all(res)
+                    for res in [
+                        list(map(self._is_int, prediction_arr))
+                        for prediction_arr in predictions
+                    ]
+                ]
+            ):
+                self.vote_type = "classificartion"
+            else:
+                self.vote_type = "regression"
+            self.vote_flag = True
+        if self.vote_type == "classification":
+            flattened_predictions = list(map(int, flattened_predictions))
+            result = self._max_vote(flattened_predictions)
+        else:
+            result = self._mean_vote(flattened_predictions)
+
+        event = {
+            "model_name": self.name,
+            "outputs": {
+                "id": events.body[list(events.body.keys())[0]].body["id"],
+                "inputs": events.body[list(events.body.keys())[0]].body["outputs"][
+                    "inputs"
+                ],
+                "prediction": result,
+            },
+            "id": events.body[list(events.body.keys())[0]].body["id"],
+        }
+
+        events.body = event
+        return events
+
+    def do_event(self, event, *args, **kwargs):
+        """handle incoming events, event is nuclio event class"""
+        start = datetime.now()
+        method = event.method or "POST"
+        if event.body and method != "GET":
+            event.body = self.parse_event(event)
+        urlpath = getattr(event, "path", "")
+
+        # if health check or "/" return Ok + metadata
+        if method == "GET" and (
+            urlpath == "/" or urlpath.startswith(self.health_prefix)
+        ):
+            setattr(event, "terminated", True)
+            event.body = self.get_metadata()
+            return event
+
+        # check for legal path prefix
+        if urlpath and not urlpath.startswith(self.url_prefix) and not urlpath == "/":
+            raise ValueError(
+                f"illegal path prefix {urlpath}, must start with {self.url_prefix}"
+            )
```

Comment:
All of this is duplicated from `BaseModelRouter`, don't duplicate, export it to an helper function there, e.g. `_pre_handle_event` and call it here

Reply:
Added as a function to `BaseModelRouter`